charts/core/src/Chart/hooks/useChart.ts

```diff
@@ -61,15 +65,71 @@ export function useChart({ theme, onChartReady }: ChartHookProps) {
     };
     window.addEventListener('resize', resizeHandler);
 
-    if (onChartReady) {
-      chartInstance.on('finished', onChartReady);
+    function onInitialRender() {
+      if (zoomable) {
```

Comment:
This was something that was discussed in #b-team last week [here](https://mongodb.slack.com/archives/C04RAE0K07R/p1732655597126739). We decided `zoomable` would turn on the "zoom effect" where one can select an area, and that `onZoomSelect` would still be needed for any zooming to occur. This is because after speaking with Gregg and Intel, it seemed there was little value in having a default zoom behavior, as zooming should always require a refetch.

The alternative would be to have `onZoomSelect` implicitly also turn on the "zoom effect", since as you noted, it won't actually zoom in on the current data and the handler is essentially required. This was a bit debated so I'm definitely open to other thoughts if this seems confusing.

Reply:
Discussed offline: After you dropped Adam and I just decided that we are going to get rid of `zoomable` altogether. We don't want to set a precedent with the pattern of implying functionality (like the click and drag functionality) based on the presence of a handler like this. However this feels like an exception to the rule due to the fact that the data is essentially always controlled and there never exists a scenario where one is useful without the other.

Will push an update shortly and re-request a review then!